cms/djangoapps/contentstore/rest_api/v0/views/tabs.py

```diff
@@ -0,0 +1,183 @@
+from openedx.core.lib.api.authentication import BearerAuthenticationAllowInactiveUser
+from openedx.core.lib.api.view_utils import DeveloperErrorViewMixin, verify_course_exists
+from ..serializers import CourseTabSerializer, CourseTabUpdateSerializer
+from ....views.tabs import get_course_tabs, update_tabs_handler
+
+
+class CourseTabSettingsView(DeveloperErrorViewMixin, APIView):
+    """
+    API view for course tabs.
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    # This is all we check here since additional access is checked when
+    # accessing the course object
+    permission_classes = (IsAuthenticated,)
+
+    def handle_exception(self, exc):
+        """Handle NotImplementedError and return a proper response for it."""
+        if isinstance(exc, NotImplementedError):
+            return self._make_error_response(400, str(exc))
+        return super().handle_exception(exc)
+
+    @apidocs.schema(
+        parameters=[apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID")],
+        responses={
+            200: CourseTabSerializer,
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    @verify_course_exists
+    def get(self, request: Request, course_id: str) -> Response:
+        """
+        Get a list of all the tabs in a course including hidden tabs.
+
+        **Example Request**
+
+            GET /api/contentstore/v0/tabs/{course_id}
+
+        **Response Values**
+
+        If the request is successful, an HTTP 200 "OK" response is returned.
+
+        The HTTP 200 response contains a list of objects that contain info
+        about each tab.
+
+        **Example Response**
+
+        ```json
+        [
+            {
+                "course_staff_only": false,
+                "is_hidden": false,
+                "is_hideable": false,
+                "is_movable": false,
+                "name": "Home",
+                "settings": {},
+                "tab_id": "info",
+                "title": "Home",
+                "type": "course_info"
+            },
+            {
+                "course_staff_only": false,
+                "is_hidden": false,
+                "is_hideable": false,
+                "is_movable": false,
+                "name": "Course",
+                "settings": {},
+                "tab_id": "courseware",
+                "title": "Course",
+                "type": "courseware"
+            },
+            ...
+        }
+        ```
+        """
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_read_access(request.user, course_key):
+            raise self.permission_denied(request)
```

Comment:
can you link me to this method? is it defined in DRF?

Reply:
Yes, it's from DRF: https://github.com/encode/django-rest-framework/blob/3875d3284e73ed4d8e36c07d9b70c1b22c9d5998/rest_framework/views.py#L169-L175

Reply:
thanks.